File: src/main/java/ru/dkolmogortsev/task/TimeEntry.java
package ru.dkolmogortsev.task;

import java.io.Serializable;
import org.hibernate.search.annotations.Indexed;
import org.joda.time.LocalDate;

/**
 * Created by dkolmogortsev on 2/25/17.
 */
@Indexed
public class TimeEntry implements Comparable<TimeEntry>, Serializable
{

    private long start;

    private long entryDate;
    private long end;
    private long duration;

    private long taskId;

    public TimeEntry(long start, long taskId)
    {
        this.start = start;
        this.entryDate = new LocalDate(start).toDate().getTime();
        this.taskId = taskId;
    }

    public long getEntryDate()
    {
        return entryDate;
    }

    public long getTaskId()
    {
        return taskId;
    }

    public long getStart()
    {
        return start;
    }

    public long getEnd()
    {
        return end;
    }

    public long getDuration()
    {
        return duration;
    }

    public void updateDuration(long duration)
    {
        this.duration = duration * 1000;
    }

    public void stop()
    {
        this.end = this.start + this.duration;
    }

    public long getId()
    {
        return taskId + start;
    }

    @Override
    public String toString()
    {
        return "TimeEntry[start:" + start + ",duration:" + duration + ']';
    }

    @Override
    public int compareTo(TimeEntry o)
    {
        return Long.compare(this.start, o.start);
    }
}

File: src/main/kotlin/ru/dkolmogortsev/task/TimeEntry.kt
package ru.dkolmogortsev.task

import org.joda.time.LocalDate
import java.io.Serializable

/**
 * Created by dkolmogortsev on 2/25/17.
 */
class TimeEntry : Comparable<TimeEntry>, Serializable {
    val start: Long
    val task: Task
    val entryDate: Long
    var end: Long = 0
        private set
    var duration: Long = 0
        private set
    val id: Long
        get() = task.id + start

    constructor(start: Long, task: Task) {
        this.start = start
        this.task = task
        this.entryDate = LocalDate(start).toDate().time
    }

    fun updateDuration(duration: Long) {
        this.duration = duration * 1000
    }

    fun stop() {
        this.end = this.start + this.duration
    }

    override fun toString(): String {
        return "TimeEntry[start:$start,duration:$duration]"
    }

    override fun compareTo(o: TimeEntry): Int {
        return java.lang.Long.compare(this.start, o.start)
    }
}
